t/run/runenv_hashseed.t

```diff
@@ -16,121 +31,193 @@ skip_all_without_config('d_fork');
+                    ($rand_bits1->[$i] // ""))
+                {
                     $bad_idx = $i;
                     last;
                 }
             }
             is($bad_idx, undef,
                 "$descr bad rand bits data index should be undef");
             if (defined $bad_idx) {
-                # we use is() to see the differing data, but this test is
-                # expected will fail - the description seems a little odd here,
-                # but since it will always fail it makes sense in context.
-                is($rand_bits_data2->[$bad_idx],$rand_bits_data1->[$bad_idx],
+                # we use is() to see the differing data, but this test
+                # is expected will fail - the description seems a little
+                # odd here, but since it will always fail it makes sense
+                # in context.
+                is($rand_bits2->[$bad_idx],$rand_bits1->[$bad_idx],
                     "$descr rand bits data is same at idx $bad_idx");
             } else {
                 pass("$descr rand bits data does not differ");
             }
         }
-        is( $err, $err2,
-            "$descr debug output was consistent between runs"
-        );
-        ################################################################################
-        # Using a different HASH_SEED
-        $seed=~s/^0x//;
-        my @chars = split //, $seed;
-        $seed = "0x" . $seed;
-
-        # increase by 1 the last digit (only)
-        $chars[-1] = sprintf( "%x", ( hex( $chars[-1] ) + 1 ) % 16 );
-        my $new_seed = "0x" . join '', @chars;
-        isnt $new_seed, $seed, "$descr got a different seed";
-        $run_opts->{PERL_HASH_SEED}= $new_seed;
-        my ( $out2, $err2 )
-            = my_runperl_and_capture( $run_opts, $print_keys );
-        isnt( $out, $out2,
-            "$descr results in different order with a different key"
-        );
     }
+
+    if ($mode eq "RANDOM") {
+        # There is a small chance we got the same result a few times
+        # even when everything is working as expected. So allow a
+        # certain number of fails.
+        ok( $random_same <= $allowed_fails,
+            "$descr_mode same key order no more than $allowed_fails times")
+            or diag(
+                "Key order was the same $random_same/$repeat times in",
+                "RANDOM mode. This test is probabilistic so if the number",
+                "is low and you re-run the tests and it does not fail",
+                "again then you can ignore this test fail.");
+    }
+
+    # There is a small chance we got the same result. So allow a certain
+    # number of fails.
+    ok( $seed_change_same <= $allowed_fails,
+        "$descr_mode same key order with munged seed no more " .
+        "than $allowed_fails times" )
+        or diag(
+            "Key order was the same $random_same/$repeat times with",
+            "a munged seed. This test is probabilistic so if the number",
```

Comment:
Given the audience for this, I suggest `s/munged/different/`.

Reply:
Ack. Willdo.